sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,589 @@
+    }
+
+    /**
+     * Get repository properties.
+     *
+     * @return repository properties.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<RepositoryProperties>> getPropertiesWithResponse() {
+        return withContext(context -> this.getPropertiesWithResponse(context));
+    }
+
+    Mono<Response<RepositoryProperties>> getPropertiesWithResponse(Context context) {
+        try {
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
+            return this.serviceClient.getPropertiesWithResponseAsync(repositoryName, context)
+                .map(res -> Utils.mapResponse(res, Utils::mapRepositoryProperties))
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Get repository properties.
+     *
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return repository properties.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<RepositoryProperties> getProperties() {
+        return this.getPropertiesWithResponse().flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * <p>Get registry artifact properties.</p>
+     *
+     * <p>This method can take in both a digest as well as a tag.<br>
+     * In case a tag is provided it calls the service to get the digest associated with it.</p>
+     * @param tagOrDigest tag or digest associated with the blob.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return registry artifact properties.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<RegistryArtifactProperties>> getRegistryArtifactPropertiesWithResponse(String tagOrDigest) {
+        return withContext(context -> this.getRegistryArtifactPropertiesWithResponse(tagOrDigest, context));
+    }
+
+    Mono<Response<RegistryArtifactProperties>> getRegistryArtifactPropertiesWithResponse(String tagOrDigest, Context context) {
+        try {
+            Mono<String> getTagMono = tagOrDigest.contains(":")
+                ? Mono.just(tagOrDigest)
+                : this.getTagProperties(tagOrDigest).map(a -> a.getDigest());
+
+            return getTagMono
+                .flatMap(tag -> this.serviceClient.getRegistryArtifactPropertiesWithResponseAsync(repositoryName, tag))
```

Comment:
I think the above mono resolves this to a `digest` right? So, should this be a `digest` instead of a tag?
```suggestion
                .flatMap(digest-> this.serviceClient.getRegistryArtifactPropertiesWithResponseAsync(repositoryName, digest))
```